src/coreclr/jit/lowerarmarch.cpp

```diff
@@ -507,14 +506,50 @@ void Lowering::LowerCast(GenTree* tree)
 
     assert(!varTypeIsSmall(srcType));
 
-    if (tmpType != TYP_UNDEF)
+    // Transform '(ulong)smallInt' or '(uint)smallInt' to 'smallInt & 0xFF..'
```

Comment:
> `(uint)smallInt`

We should never see these in optimized code, they're nops. I think this would be better illustrated with the IR transform performed, something like: `CAST(long <- CAST(small type <- X)) => AND(int/long X, 0xFF/0xFFFF)`, noting too that this creates "illegal" IR (`AND(int, long) => long`).